app-misc/rpick/rpick-0.4.0.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Auto-Generated by cargo-ebuild 0.2.0
+
+EAPI=7
+
+CRATES="
+ansi_term-0.11.0
+arrayref-0.3.5
+arrayvec-0.4.12
+atty-0.2.13
+autocfg-0.1.7
+backtrace-0.3.40
+backtrace-sys-0.1.32
+base64-0.10.1
+bitflags-1.2.1
+blake2b_simd-0.5.8
+byteorder-1.3.2
+cc-1.0.46
+cfg-if-0.1.10
+clap-2.33.0
+cloudabi-0.0.3
+constant_time_eq-0.1.4
+crossbeam-utils-0.6.6
+dirs-1.0.5
+dtoa-0.4.4
+failure-0.1.6
+failure_derive-0.1.6
+fuchsia-cprng-0.1.1
+heck-0.3.1
+lazy_static-1.4.0
+libc-0.2.65
+linked-hash-map-0.5.2
+nodrop-0.1.14
+proc-macro2-0.4.30
+proc-macro2-1.0.6
+quote-0.6.13
+quote-1.0.2
+rand-0.6.5
+rand_chacha-0.1.1
+rand_core-0.3.1
+rand_core-0.4.2
+rand_hc-0.1.0
+rand_isaac-0.1.1
+rand_jitter-0.1.4
+rand_os-0.1.3
+rand_pcg-0.1.2
+rand_xorshift-0.1.1
+rdrand-0.4.0
+redox_syscall-0.1.56
+redox_users-0.3.1
+rpick-0.4.0
+rust-argon2-0.5.1
+rustc-demangle-0.1.16
+serde-1.0.102
+serde_derive-1.0.102
+serde_yaml-0.8.11
+strsim-0.8.0
+structopt-0.2.18
+structopt-derive-0.2.18
+syn-0.15.44
+syn-1.0.7
+synstructure-0.12.1
+textwrap-0.11.0
+unicode-segmentation-1.5.0
+unicode-width-0.1.6
+unicode-xid-0.1.0
+unicode-xid-0.2.0
+vec_map-0.8.1
+winapi-0.3.8
+winapi-i686-pc-windows-gnu-0.4.0
+winapi-x86_64-pc-windows-gnu-0.4.0
+yaml-rust-0.4.3
+"
+
+inherit cargo
+
+DESCRIPTION="Helps you pick items from a list by various algorithms."
+HOMEPAGE="https://gitlab.com/bowlofeggs/rpick"
+SRC_URI="$(cargo_crate_uris ${CRATES})"
+RESTRICT="mirror"
+LICENSE="GPL-3"
```

Comment:
Here comes the bad news: You need to write all licenses used by the bundled crates here as well. I am not sure how to do that with rust packages, but Go that works similarily can use `glice` or `go-license-detector` for example. You may need to search for some tools that can automate this check for you. 

And currently yes, you need to re-check it every version bump.

Reply:
This tool looks helpful for doing this:

https://crates.io/crates/cargo-license

I'll give that a whirl and see how well it works.

Reply:
OK, it looks like ```cargo license``` is going to be helpful after all:

```
cargo license
Apache-2.0/MIT (33): rand, nodrop, rand_xorshift, yaml-rust, structopt-derive, rand_hc, backtrace, backtrace-sys, winapi, rand_chacha, bitflags, rustc-demangle, structopt, rand_core, serde_yaml, rand_os, unicode-width, unicode-xid, unicode-segmentation, cfg-if, rand_isaac, quote, arrayvec, rand_pcg, cc, rand_core, dtoa, vec_map, winapi-i686-pc-windows-gnu, winapi-x86_64-pc-windows-gnu, proc-macro2, autocfg, linked-hash-map
BSD-2-Clause (1): cloudabi
CC0-1.0 (1): constant_time_eq
GPL-3.0 (1): rpick
ISC (1): rdrand
MIT (12): redox_termios, clap, redox_syscall, scoped_threadpool, textwrap, redox_users, strsim, argon2rs, termion, synstructure, atty, ansi_term
MIT OR Apache-2.0 (12): blake2-rfc, failure, serde, heck, rand_jitter, numtoa, failure_derive, libc, serde_derive, dirs, syn, dirs-sys
N/A (1): fuchsia-cprng
```

That one that says N/A is kind of weird though. It's actually not included in the list of crates generated by ```cargo-ebuild```, nor is it downloaded or compiled when I run ```cargo build``` by hand. So I don't think we have to worry about it. For the record, it's this:

https://crates.io/crates/fuchsia-cprng

Their code repo is here:

https://fuchsia.googlesource.com/fuchsia/+/master/garnet/public/rust/fuchsia-cprng

If you look in the various files, they all say this:

```
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
```

Oddly, there is no ```LICENSE``` file in the repository. But again, since it isn't used by rpick, I don't think we need to worry about it.

Thus, I think I can do this:

```
LICENSE="GPL-3 Apache-2.0 BSD-2 CC0-1.0 ISC MIT"
```